mindsdb/api/http/namespaces/agents.py

```diff
@@ -164,23 +164,33 @@ def put(self, project_name, agent_name):
         agents_controller = AgentsController()
 
         try:
-            existing_agent = agents_controller.get_agent(agent_name, project_name=project_name)
-        except ValueError:
+            existing_agent_record = agents_controller.get_agent(agent_name, project_name=project_name)
+        except (ValueError, EntityNotExistsError):
             # Project must exist.
             return http_error(
                 HTTPStatus.NOT_FOUND,
                 'Project not found',
                 f'Project with name {project_name} does not exist'
             )
+        if existing_agent_record is None:
+            raise Exception
 
         agent = request.json['agent']
         name = agent.get('name', None)
         model_name = agent.get('model_name', None)
         skills_to_add = agent.get('skills_to_add', [])
         skills_to_remove = agent.get('skills_to_remove', [])
+        skills_to_rewrite = agent.get('skills', [])
         provider = agent.get('provider')
         params = agent.get('params', None)
 
+        if len(skills_to_rewrite) > 0 and (len(skills_to_remove) > 0 or len(skills_to_add) > 0):
+            return http_error(
+                HTTPStatus.BAD_REQUEST,
+                'Wrong arguments',
+                "'skills_to_rewrite' and 'skills_to_add' (or 'skills_to_remove') cannot be used at the same time"
+            )
+
```

Comment:
this block can be removed as it is checked in [controller](https://github.com/mindsdb/mindsdb/pull/10093/files#diff-929cbbd4d798cd3b3622b0bf44b716396b376f11f6e18a68aaa2a0824d8260bfR245)